polygeist/tools/cgeist/Lib/CGStmt.cc

```diff
@@ -997,6 +997,49 @@ ValueCategory MLIRScanner::VisitSwitchStmt(clang::SwitchStmt *stmt) {
   return nullptr;
 }
 
+ValueCategory MLIRScanner::VisitBinAssign(BinaryOperator *e) {
+  LLVM_DEBUG({
+    llvm::dbgs() << "VisitBinAssign: ";
+    e->dump();
+    llvm::dbgs() << "\n";
+  });
+  auto rhs = Visit(e->getRHS());
+  auto lhs = Visit(e->getLHS());
+
+  assert(lhs.isReference);
```

Comment:
It seems the assertion allows for assignment expressions where the LHS is a reference. Is it possible the LHS is a pointer dereference or some other type of expression. `VisitBinAssign` receives a BinaryOperator expression so in principle the LHS (and RHS) expressions could be complex.

Reply:
I haven't explored that in particular, but I will address all the flaws I find in a different PR (this one just fixes eval order and return value).